examples/csv2argus.py

```diff
@@ -0,0 +1,261 @@
+        csvr = csv.reader(input)
+        # var for column names from first row in result
+        cols = None
+        data = []
+        for row in csvr:
+            logging.debug("Got row: %s", row)
+            # Assign cols from the first row, the header in the CSV
+            if not cols:
+                cols = row
+                continue
+            # Append rows to data
+            data.append(dict(zip(cols, row)))
+
+        if not opts.quiet:
+            logging.info("Total result count: %s", len(data))
+
+        # dictionary var to hold metrics
+        m_dict = {}
+        # for loop to populate m_dict
+        for row in data:
+            # abort without timestamp
+            try:
+                ts = row[opts.timestampcolumn] and to_gmt_epoch(row[opts.timestampcolumn])
+            except KeyError:
+                logging.error("Error: Timestamp not found: %s", row)
+                return None
+
+            # create final scope, substitute keys into the scope
+            rowScope = opts.argusscope
+
+            # abort without arguskeys
+            if keyNames:
+                for keyName in keyNames:
+                    logging.debug("Subbing keyName " + keyName + " with value " + str(row[keyName]))
+                    try:
+                        rowScope = rowScope.replace("{" + keyName + "}", row[keyName])
+                        logging.debug("New rowScope = " + rowScope)
+                    except KeyError:
+                        logging.error("Error: Specified arguskeys not found: %s", row)
+                        return None
+
+            # create tags
+            tag_dict = {}
+            if tagNames:
+                for tagName in tagNames:
+                    logging.debug("Setting tag pair for name: " + tagName + " and value: " + str(row[tagName]))
+                    # abort without argustags
+                    try:
+                        tag_dict[tagName] = row[tagName]
+                    except KeyError:
+                        logging.error("Error: Specified tags not found: %s", row)
+                        return None
+
+            # create metrics and datapoints
+            for col in metricNames:
+
+                # abort without argusmetrics
+                try:
+                    val = row[col]
+                except KeyError:
+                    # Key is not present
+                    logging.error("Error: Specified metrics not found: %s", row)
+                    return None
+
+                # abort for non-numeric metrics
+                try:
+                    float(val)
+                except:
+                    logging.error("Error: Non-numeric metric found: %s", row)
+                    return None
+
+                # cast str to number
+                if "." in val:
+                    val = float(val)
+                else:
+                    val = int(val)
```

Comment:
I know this came from `splunk2argus.py`, but it seems like I didn't pay attention to this code before as the lines from 192 to 210 can be done simpler and cleaner like this:

```python
if val.isdigit():
    val = int(val)
else:
    try:
        val = float(val)
    except:
         logging.error("Error: Non-numeric metric found: %s", row)
         return None
```